Sources/Protocols/Animatable/Animatable.swift

```diff
@@ -92,7 +92,24 @@ public extension Animatable where Self: UIView {
     let completion = {
       promise.animationCompleted()
     }
-    switch animation ?? animationType {
+    doAnimation(animation ?? self.animationType, configuration: configuration, completion: completion)
+  }
+
+  /**
+   `autoRunAnimation` method, should be called in layoutSubviews() method
+   */
+  func autoRunAnimation() {
+    if autoRun {
+      autoRun = false
+      animate(animationType)
+    }
+  }
+}
+
+fileprivate extension UIView {
```

Comment:
Since it is `private`, that's fine to extend `UIView`, then we can share it with `UIBarItem`'s subview.